MoltenVK/MoltenVK/Commands/MVKCmdPipeline.mm

```diff
@@ -65,8 +65,36 @@
 
 #if MVK_MACOS
     // Calls below invoke MTLBlitCommandEncoder so must apply this first
-	if ( !(_memoryBarriers.empty() && _imageMemoryBarriers.empty()) ) {
-		[cmdEncoder->_mtlRenderEncoder textureBarrier];
+	if ( [cmdEncoder->_mtlRenderEncoder respondsToSelector: @selector(memoryBarrierWithScope:afterStages:beforeStages:)] ) {
```

Comment:
I recognize that this is a very localized test...but I would prefer that any Metal feature availability be enabled in `MVKPhysicalDeviceMetalFeatures` in `MVKDevice`...so that we have a single public record of available Metal features.

Can you add a `memoryBarrier` (or equivalent) member to `MVKPhysicalDeviceMetalFeatures`, and test for it here, please?

You don't need to increment `VK_MVK_MOLTENVK_SPEC_VERSION` beyond the `20` it became in PR #591.

Thanks.

(I recognize that we do have some `respondToSelector:` tests in the Metal class extensions...but is localized to the class itself).

Reply:
Thanks!